pdr_backend/analytics/predictoor_dashboard/dash_components/util.py

```diff
@@ -53,6 +53,35 @@ def get_predictoors_data_from_db(lake_dir: str):
     )
 
 
+@enforce_types
+def get_feed_ids_based_on_predictoors_from_db(
+    lake_dir: str, predictoor_addrs: List[str]
+):
+    # Constructing the SQL query
+    query = f"""
+        SELECT LEFT(ID, POSITION('-' IN ID) - 1) AS feed_addr
+        FROM {Payout.get_lake_table_name()}
+        WHERE ID IN (
+            SELECT MIN(ID)
+            FROM {Payout.get_lake_table_name()}
+            WHERE (
+                {" OR ".join([f"ID LIKE '%{item}%'" for item in predictoor_addrs])}
+            )
+            GROUP BY LEFT(ID, POSITION('-' IN ID) - 1)
+        );
+    """
+
+    # Execute the query
+    resp = _query_db(
+        lake_dir,
+        query,
+    )
+
+    # Extract feed addresses from the response
+    feed_ids = [obj["feed_addr"] for obj in resp]
```

Comment:
I'm not sure about this. In theory, there should be a way to retrieve the data as a list instead of a DataFrame. Or to select the column as a list directly, instead of iterating through the objects. I'll investigate to help.

Reply:
maybe with `unnest`? https://duckdb.org/docs/sql/query_syntax/select.html#unnest-function

Reply:
or `list()`

Returns a list of all regions in order of the amount column:

SELECT list(region ORDER BY amount DESC)
FROM sales;

https://duckdb.org/docs/sql/functions/aggregates.html